### PR TITLE
[test-suite] Error when calling test-suite without a full build.

### LINKED ISSUE #14887: Unable to run suite tests locally
After trying many permutations, I'm unable to usefully run the test suite locally (Ubuntu on WSL).

I did `./configure -profile devel` and `make -f Makefile.dune world` to set up.

Then, following the instructions in `build-system.dune.md`:
1. `make -C test-suite` runs the tests, but it appears that every test fails with `Error: Unable to locate library Prelude with prefix Coq.` (`COQLIB` not defined).
2. Running the same command after ```export COQLIB=`pwd` ``` in the top directory of the source tree gives the same messages
3. Trying `dune runtest`, many many tests get `Error: Cannot find plugins directory`
4. `COQLIB` pointing to `.../_build/default` is not different

Also, how can I run a single test presently?

### PR DESCRIPTION
**Pull request by ejgallego**
Fixes #14887

### CODE REVIEW

File: test-suite/Makefile [hunk 1]
```diff
@@ -57,11 +56,22 @@ export CAML_LD_LIBRARY_PATH:=$(shell echo $(COQPREFIX)/lib/stublibs):$(CAML_LD_L
...
 COQLIB?=
+
 ifeq ($(COQLIB),)
+# This should guard againt the COQLIBINSTALL directory not existing,
+# however, it doesn't work, defying my knowledge about makefiles.
+ifneq ($(wildcard $(COQLIBINSTALL)),)
 	COQLIB := $(shell ocaml ocaml_pwd.ml $(COQLIBINSTALL))
+else
+	COQLIB := $(shell ocaml ocaml_pwd.ml $(COQLIBINSTALL))
+endif
```
Comment (SkySkimmer) [inline test-suite/Makefile:67]: What's the point of this?

**Review by SkySkimmer — DISMISSED** [review]
This isn't a warning, it's an error

File: test-suite/Makefile (hunk 1 above)
Comment (ejgallego) [reply test-suite/Makefile:67]: To have someone that understands make to fix it some day.

Comment (ejgallego) [reply test-suite/Makefile:67]: I'd rather remove the whole ocaml_pwd stuff, but that's a different story. The use case for the if is as follows:
- `ocaml_pwd` requires the path to exists, so if the user calls `make -C test-suite` before the path was created, and error follows.

Comment (SkySkimmer) [reply test-suite/Makefile:67]: >To have someone that understands make to fix it some day.

But what is it supposed to do? it seems like there should be something else in the `else` branch.

Comment (ejgallego) [reply test-suite/Makefile:67]: The `else` branch should be a noop, this is guarding against calling `ocaml_pwd` with a non existing path.

Better, we should get rid of `ocaml_pwd`, but that's for another day.

Comment (ejgallego) [reply test-suite/Makefile:67]: I have no idea, but I was unable to have that conditional work fine.

**Comment by ejgallego** [comment]
> This isn't a warning, it's an error

Corrected.

**Comment by ejgallego** [comment]
The question by @SkySkimmer on how to fix the make code is still on, however note that it is not a regression from the current state of the facts, so IMHO we could go ahead.

File: test-suite/Makefile [hunk 2]
```diff
@@ -57,11 +56,22 @@ export CAML_LD_LIBRARY_PATH:=$(shell echo $(COQPREFIX)/lib/stublibs):$(CAML_LD_L
 
 # COQLIB is an env variable so no quotes
 COQLIB?=
+
 ifeq ($(COQLIB),)
+# This should guard againt the COQLIBINSTALL directory not existing,
+# however, it doesn't work, defying my knowledge about makefiles.
+ifneq ($(wildcard $(COQLIBINSTALL)),)
```
Comment (SkySkimmer) [inline test-suite/Makefile:63]: COQLIBINSTALL doesn't exist because it is quoted

File: test-suite/Makefile [hunk 3]
```diff
@@ -148,8 +158,29 @@ PREREQUISITELOG = $(addsuffix .log,$(wildcard prerequisite/*.v)) .csdp.cache
 .DELETE_ON_ERROR:
 .PHONY: all run clean $(SUBSYSTEMS)
 
+ifeq ($(COQ_NOT_CONFIGURED),true)
+all:
+	@echo ""
+	@echo "Coq has not been configured, do any of: "
+	@echo "  - make -f Makefile.dune wolrd"
```
Comment (SkySkimmer) [inline test-suite/Makefile:165]:
```suggestion
	@echo "  - make -f Makefile.dune world"
```

File: test-suite/Makefile [hunk 4]
```diff
@@ -148,8 +158,29 @@ PREREQUISITELOG = $(addsuffix .log,$(wildcard prerequisite/*.v)) .csdp.cache
...
+	@echo "  - ./configure -profile devel && make -j NJOBS world    "
```
Comment (SkySkimmer) [inline test-suite/Makefile:166]:
```suggestion
	@echo "  - ./configure -profile devel && make -j NJOBS world"
```

File: test-suite/Makefile [hunk 5]
```diff
@@ -148,8 +158,29 @@ PREREQUISITELOG = $(addsuffix .log,$(wildcard prerequisite/*.v)) .csdp.cache
 .DELETE_ON_ERROR:
 .PHONY: all run clean $(SUBSYSTEMS)
 
+ifeq ($(COQ_NOT_CONFIGURED),true)
+all:
+	@echo ""
+	@echo "Coq has not been configured, do any of: "
```
Comment (Alizter) [inline test-suite/Makefile:164]:
```suggestion
	@echo "Coq has not been configured, please run one of the following: "
```
We only want one to be run right?

**Review by SkySkimmer — CHANGES_REQUESTED** [review]
Waiting for the COQLIBINSTALL code to be fixed now that we know what the problem is.

File: test-suite/Makefile (hunk 2 above)
Comment (ejgallego) [reply test-suite/Makefile:63]: Oh O_O, ok, hack added.

File: test-suite/Makefile [hunk 6]
```diff
@@ -57,11 +56,20 @@ export CAML_LD_LIBRARY_PATH:=$(shell echo $(COQPREFIX)/lib/stublibs):$(CAML_LD_L
 
 # COQLIB is an env variable so no quotes
 COQLIB?=
+
 ifeq ($(COQLIB),)
+# This should guard againt the COQLIBINSTALL directory not existing,
+# however, it doesn't work, defying my knowledge about makefiles.
```
Comment (SkySkimmer) [inline test-suite/Makefile:62]: outdated comment?

Comment (ejgallego) [reply test-suite/Makefile]: Indeed

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Pull request merged**